scripts/simulator/daoActionsGenerator.js

```diff
@@ -0,0 +1,331 @@
+
+    if (!new BN(campID).eq(new BN(0))) {
+      console.log('already have campID ' + campID + ' for epoch' + startEpoch)
+      result.isValid = false
+      result.msg = 'validateParams: already had brr campaign for this epoch'
+    } else {
+      if (lastRewardInBps.gt(new BN(7000))) {
+        result.isValid = false
+        result.msg = 'validateParams: rebate + reward must be smaller then BPS'
+      } else {
+        result.msg = 'create new brr campaign at epoch ' + startEpoch
+      }
+    }
+  } else {
+    if (!validOption) {
+      result.options = [new BN(0), new BN(1), new BN(2)]
+      result.isValid = false
+      result.msg = 'validateParams: general campaign option is 0'
+    }
+  }
+  return result
+}
+
+module.exports.getEpochNumber = getEpochNumber
+function getEpochNumber (epochPeriod, startTime, timestamp) {
+  if (new BN(timestamp).lt(new BN(startTime))) return new BN(0)
+  return new BN(timestamp)
+    .sub(new BN(startTime))
+    .div(new BN(epochPeriod))
+    .add(new BN(1))
+}
+
+// random select a campaignID from current epoch or next epoch
+module.exports.genCancelCampaign = async (daoContract, currentBlockTime, epoch) => {
+  let campaigns = await daoContract.getListCampaignIDs(epoch)
+  campaigns = campaigns.concat(await daoContract.getListCampaignIDs(epoch.add(new BN(1))))
+  if (campaigns.length == 0) {
+    console.log(`there is no campain in epoch ${epoch} and epch ${epoch.add(new BN(1))} to cancel`)
+    return undefined
+  }
+  let campaignID = campaigns[genRandomNumber(campaigns.length)]
+  let campaignDetails = await daoContract.getCampaignDetails(campaignID)
+  if (campaignDetails.startTimestamp <= currentBlockTime) {
+    return {
+      blockTime: currentBlockTime,
+      isValid: false,
+      msg: 'cancelCampaign: campaign already started',
+      campaignID
+    }
+  }
+
+  return {
+    blockTime: currentBlockTime,
+    isValid: true,
+    msg: `cancel Campaign ${campaignID} in epoch ${epoch}`,
+    campaignID
+  }
+}
+
+// random select a campaign ID from this epoch and select random option
+module.exports.genVote = async (daoContract, currentBlockTime, epoch, stakers) => {
+  let campaigns = await daoContract.getListCampaignIDs(epoch)
+  let staker = stakers[genRandomNumber(stakers.length)]
+  if (campaigns.length == 0) {
+    console.log(`there is no campain in epoch ${epoch} to vote`)
+    if (genRandomNumber(100) >= 90) {
+      numCampaign = await daoContract.numberCampaigns()
+      return {
+        staker,
+        campaignID: new BN(numCampaign).add(new BN(2)),
+        option: new BN(1),
+        isValid: false,
+        msg: "vote: campaign doesn't exist"
+      }
+    }
+    // returns undefined so simulater will genVote instead
+    return undefined
+  }
+  let campaignID = campaigns[genRandomNumber(campaigns.length)]
+  let campaignDetails = await daoContract.getCampaignDetails(campaignID)
+
+  let option = genRandomNumber(campaignDetails.options.length + 2)
```

Comment:
can try to add more invalid values.
but give it low weight.

so maybe half votes in valid range and other
half randomised for total number range.

Reply:
I add invalid option is random. 
However, creating them with small possibility